mysql-test/suite/galera_sr/t/MDEV-27553.test

```diff
@@ -5,29 +5,74 @@
 --source include/galera_cluster.inc
 --source include/have_debug.inc
 
+--let $node_1=node_1
+--let $node_2=node_2
+--source suite/galera/include/auto_increment_offset_save.inc
+
+--connect node_2a, 127.0.0.1, root, , test, $NODE_MYPORT_2
+
 CREATE TABLE t1 (f1 INTEGER PRIMARY KEY);
 
---connection node_1
---let $wsrep_cluster_address_orig = `SELECT @@wsrep_cluster_address`
+--connection node_2
 SET SESSION wsrep_trx_fragment_size=1;
 START TRANSACTION;
 INSERT INTO t1 VALUES (1);
-# This will result in failure to remove fragments
-# from streaming log, in the following ROLLBACK.
+SELECT COUNT(*) `Expect 1` FROM mysql.wsrep_streaming_log;
+
+#
+# Issue ROLLBACK and make sure it fails to clean up
+# the streaming log. Failure to remove fragments
+# results in apply failure of the rollback fragment.
+# The node should disconnect from the cluster.
+#
 SET @@global.debug_dbug="+d,ha_index_init_fail";
 ROLLBACK;
 
---connection node_2
+
+#
+# Expect the cluster to shrink
+#
+--connection node_1
+SET SESSION wsrep_sync_wait = 0;
+--let $wait_condition = SELECT VARIABLE_VALUE = 1 FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'wsrep_cluster_size'
+--source include/wait_condition.inc
+
+#
+# ROLLBACK should clean up the streaming log just fine in node 1
+#
 SELECT COUNT(*) `Expect 0` FROM mysql.wsrep_streaming_log;
 
 
---connection node_1
+#
+# Expect the failure on ROLLBACK to leave a entry in streaming log
+#
+--connection node_2
 SET @@global.debug_dbug="";
+SET SESSION wsrep_sync_wait = 0;
 SELECT COUNT(*) `Expect 1` FROM mysql.wsrep_streaming_log;
+# Expect node to be disconnected
```

Comment:
Delivering disconnected view may be delayed. Better to add wait condition for disconnected cluster status here.

Reply:
Fixed.